### PR TITLE
Add new search parameters highlightPreTag, highlightPostTag and cropMarker

### PR DESCRIPTION
**Pull request by alallema**
As per https://github.com/meilisearch/meilisearch/issues/2214 new search parameters are introduced: 

- `highlightPreTag`
- `highlightPostTag`
- `cropMarker`

### CODE REVIEW

File: index_search_test.go [hunk 1]
```diff
@@ -164,13 +164,16 @@ func TestIndex_Search(t *testing.T) {
 				query:  "to",
 				request: SearchRequest{
 					AttributesToCrop: []string{"title"},
-					CropLength:       7,
+					CropLength:       2,
```
Comment (bidoubiwa) [inline index_search_test.go:167]: is this indentation normal? It feels like it is one space different with [

File: index_search_test.go [hunk 2]
```diff
@@ -179,6 +182,87 @@ func TestIndex_Search(t *testing.T) {
...
+					},
+				},
+				NbHits:           2,
+				Offset:           0,
+				Limit:            1,
+				ExhaustiveNbHits: false,
+			},
+		},
+		{
+			name: "TestIndexSearchWithAttributeToHighlightWithCustomPreAndPostTag",
```
Comment (bidoubiwa) [inline index_search_test.go:239]:
```suggestion
			name: "TestIndexSearchWithCustomPreAndPostHighlightTags",
```

File: index_search.go [hunk 3]
```diff
@@ -38,6 +38,15 @@ func (i Index) Search(query string, request *SearchRequest) (*SearchResponse, er
 	if request.CropLength != 0 {
 		searchPostRequestParams["cropLength"] = request.CropLength
 	}
+	if request.CropMarker != "" {
+		searchPostRequestParams["cropMarker"] = request.CropMarker
+	}
+	if request.HighlightPreTag != "" {
```
Comment (brunoocasali) [inline index_search.go:44]: There is a way to automate this kind of thing, like serialize somehow?

File: index_search_test.go [hunk 4]
```diff
@@ -100,7 +100,7 @@ func TestIndex_Search(t *testing.T) {
 			want: &SearchResponse{
 				Hits: []interface{}{
 					map[string]interface{}{
-						"book_id": float64(1), "title": "Alice In Wonderland",
+						"book_id": float64(123), "title": "Pride and Prejudice",
```
Comment (brunoocasali) [inline index_search_test.go:103]: Can you give some idea about this `float64(...)`? Why do you need to convert it to float?

File: index_search.go (hunk 3 above)
Comment (alallema) [reply index_search.go:44]: I would love it! I didn't find any other ways

File: index_search_test.go (hunk 4 above)
Comment (alallema) [reply index_search_test.go:103]: This is the default behavior of the deserializer in golang you can read in the [documentation](https://pkg.go.dev/encoding/json#Unmarshal)

> To unmarshal JSON into an interface value, Unmarshal stores one of these in the interface value:
bool, for JSON booleans
float64, for JSON numbers
string, for JSON strings
[]interface{}, for JSON arrays
map[string]interface{}, for JSON objects
nil for JSON null

And by default, every number declare are `int` so in:
```
	map[string]interface{}{
		"book_id": 1,
	}
```
`book_id` will be an int that is why I convert it. I didn't find a better way to do it.

**Review by brunoocasali — APPROVED** [review]
Awesome! (go tests are a little bit harder to read for me yet, but I think they are ok) 👯‍♂️

**Comment by alallema** [comment]
bors merge

**Pull request merged**